app/src/main/java/com/github/wnder/picture/FirebasePicturesDatabase.java

```diff
@@ -199,6 +198,16 @@ public FirebasePicturesDatabase(){
         return cf;
     }
 
+    @Override
+    public CompletableFuture<Bitmap> getMapSnapshot(String uniqueId) {
+        throw new IllegalStateException("This method is only available on the local database");
+    }
+
+    @Override
+    public CompletableFuture<Location> getUserGuess(String uniqueId) {
+        throw new IllegalStateException("This method is only available on the local database");
+    }
+
```

Comment:
These methods should return a completable future completed exceptionnaly instead of throwing an exception.
Also, getting a user guess is possible though it needs more steps because we need the unique ID and the user name, then find the user in the list of guesses for this picture on the DB. We should implement it (maybe not it this PR because it's not completely necessary for the feature). Also it could allow to implement the getMapSnapshot because from what I understand we can generate the map snapshot from 2 point (when we are online).